### ISSUE TITLE
Allow to hide the thought process of "thinking" models in the assistant panel

### ISSUE THREAD

**Issue by bnn1** [zed-industries/zed#22373]
Summary: Allow to hide the thought process of "thinking" models like o1 in the assistant panel

Currently, when using thinking models like `gemini-2.0-flash-thinking-exp-1219`, the whole thought process is dumped on user along with the actual response to the user's query. 

Hiding thought process into collapsible section would be preferred when dealing with thinking models along with the indication that the model is "thinking". 

### Zed Version and System Specs

Zed: v0.166.2 (Zed)
OS: Linux X11 arcolinux unknown
Memory: 30.9 GiB
Architecture: x86_64
GPU: Intel(R) Iris(R) Xe Graphics (ADL GT2) || Intel open-source Mesa driver || Mesa 24.3.2-arch1.1

### If applicable, add mockups / screenshots to help present your vision of the feature

Current behavior:
![image](https://github.com/user-attachments/assets/47fc9e47-5ad5-4c91-9962-7b7eaa6c5624)
![image](https://github.com/user-attachments/assets/08cf33cf-2b07-4f09-a8ec-8c02635fa5a7)

Desired behavior:
![image](https://github.com/user-attachments/assets/b62d72a0-975e-4282-ba5f-66c99627c0da)

**Comment by bnn1** [comment]
@gyzerok The "Thinking process" collapsible section you observe on the "Desired behavior" screenshot was photoshoped, sorry for the confusion 😅

**Comment by gyzerok** [comment]
@bnn1 yeah, I know. I've just reported my use case as well, so maintainers have more info to word with. It just so happened that your section name and mine appears to be very close

**Comment by nathom** [comment]
This would be great since deepseek just released the `deepseek-reasoner` API!

**Comment by RustoMCSpit** [comment]
> I am not using a "thinking" model, however am currently working on a prompt that would force model to think. Was expecting to be able to hide this thought process into collapsible section like on the screenshot, but it didn't work out.
> 
> Would love to see this use case supported as well.
> 
> ![image](https://github.com/user-attachments/assets/f7ecf2e3-efa0-4caf-a9a7-58366e1c357f)

it looks like Zed will need config options as there is no standardised tag. youre using <summary>, deepseek has <think>, etc.

**Comment by lucascherzer** [comment]
FYI: The chat title also does not strip the thinking process.
It might also be worth considering to fall back on small models for title generation, as this task is rather simple in most cases and choosing small models for this task is more efficient.

**Comment by tbiehn** [comment]
Thinking models also impact (break) inline edit in addition to showing up in assistant titles... Which might be rolled into this issue, but those two issues are more of a 'bug' than this cosmetic request.

**Comment by popovpsk** [comment]
hello there. last month many models were being launched in open source, like phi4, qwen3, granite3.3, i think some model is releasing right now, while i write this message. Looks like the priority of this issue is growing, because Zed experience with all these new models is pretty bad.

**Comment by jeremyh** [comment]
Yes — open models are increasingly including the `<think>` tag. Simply being able to collapse them would be a great improvement to readability.

**Comment by ViViDboarder** [comment]
As @tbiehn mentions, thinking models, like Qwen3, also break agentic editing. I asked it to update some files and each file has the` `<think>` content written into it as well, making it pretty much useless.

**Comment by captainvera** [comment]
Same as the people above. I wanted to try using Qwen3 with the newest agentic mode but it is basically unusable since it writes the whole thinking process into the file.

**Comment by innotime32** [comment]
**It would be nice if Zed supported the latest/best highest ranking offline coding/reasoning models, which all use think tags in their output response.**

On Ollama: 1.2M downloads for Qwen3 ... 43M downloads for Deepseek R1 variants ... seems high priority to support these models! **These models use think tags in their output.**

 **Inline assist** needs to display "Thinking..." while the model outputs the think tag, after the tag completes, this needs to be removed from the output when inserting the code and rendering the inline diff). otherwise it is unusable as the think tag gets inserted into the code/diff.

    **In the Agent Panel** hide do something similar: hide thinking by default, have a drop-down to show/hide the think tag (thought process)... otherwise you are seeing 50%+ irrelevant information in the panel.

**This issue could be higher priority IMO.
Otherwise we have to another IDE or older non-reasoning coding models.
Zed is awesome!! let's support the latest coding/reasoning models ⚡️💪✨❤️**

**Comment by nathanjcollins** [comment]
I'm trying to use qwen3 and it actually ends up responding to it's own <think> response, so I never get an answer to my original question.

**Comment by mikhail-shevtsov-wiregate** [comment]
I think this ticket can be closed because both `ollama` and `zed` do support `<think>` tags now and it works rather good. I use them both daily.

**Comment by ghost91-** [comment]
It's still an issue when using models that use `<think>` tags via another API than Ollama, e.g. via an OpenAI compatible API. So I would suggest to keep it open. Since there is no dedicated API support for thinking in these cases, I don't think there is another option than simply parsing the output for `<think>` tags. I believe, that's what Continue is doing, for example.

**Comment by adsick** [comment]
Recently a couple of capable OpenAI's models dropped and they don't support "not thinking". As people mentioned, this breaks code suggestions as "thinking" gets included into them.

upd: seems like it was "fixed" on LM-studio's side, I guess they simply don't stream "thinknig" now.

**Comment by lee-b** [comment]
Would really suggest that devs prioritise this. It's basic stuff, in August 2025.

**Comment by aarononeal** [comment]
Seems this is not just cosmetic. `<think>` responses also break tool calling.

**Comment by morgankrey** [comment]
Given that 
- we collapse thinking blocks in the current agent panel
- the original issue is in reference to a surface that's been iterated on substantially since release
Closing this issue, and if there are provider-specific issues that crop up or remain that are related, please open a new issue and include this one as context.
